### PR TITLE
Remove use of artifacts-api.elastic.co and its client

### PR DESCRIPTION
**Pull request by rdner**
This includes removal of
TestStandaloneDowngradeToSpecificSnapshotBuild integration test.

Discussed with @pchila and @cmacknz 

1. Independent agent patch releases will be ready very soon.
2. We are likely to handle downgrades in a different way than “upgrade to lower version” in the actual product.
3. This test fails our integration tests sue to instability of `artifacts-api.elastic.co` and that's the only code that is still using this API.

Closes https://github.com/elastic/elastic-agent/issues/4347

### CODE REVIEW

File: pkg/testing/tools/artifacts_api.go
Comment (pchila) [inline pkg/testing/tools/artifacts_api.go:1]: I think that parts of this are used in magefile in relation of parsing build manifests (mostly datamodel)
For example https://github.com/elastic/elastic-agent/blob/db40ac25757dc35cc61f92a4fea3e8864a97d1cb/magefile.go#L486

Comment (rdner) [reply pkg/testing/tools/artifacts_api.go:1]: Okay, I'll move structures then.

Comment (rdner) [reply pkg/testing/tools/artifacts_api.go:1]: https://github.com/elastic/elastic-agent/pull/5050/commits/2a2ba2ea59c3da8ca57017af37e8a09cc4bc244c

**Comment by rdner** [comment]
@cmacknz are we still doing this?

**Review by cmacknz — APPROVED** [review]
I'm fine with it, this leaves some infrequently and not critical path code without a test, but eliminates the most frequent test failure.

**Pull request merged**